server/endpoints/chat/index.js

```diff
@@ -6,49 +6,52 @@ const router = new express.Router();
 
 router.get('/chat/:id', function(req, res) {
   const currentUser = req.user;
-  
-  currentUser.deepPopulate([
-    'messages',
-    'messages.sender',
-    ], (err, user) => {
-    if (err) {
-      throw err;
-    }
-    User.findById(req.params.id, (err, usr) => {
-      if(!usr) {
-        res.redirect('/chat');
-      } else {
-        Message.find({receiver: currentUser.id, sender: usr.id}, (err, messages) => {
-          messages.filter((message) => {return !message.isSeen})
-            .map((message) => {
-              message.isSeen = true;
-              message.save((err) => {
-                if(err) throw err;
+  if(req.params.id === currentUser.id) {
+    res.redirect('/chat');
+  } else {
+    currentUser.deepPopulate([
+      'messages',
+      'messages.sender',
+      ], (err, user) => {
+      if (err) {
+        throw err;
+      }
+      User.findById(req.params.id, (err, usr) => {
+        if(!usr) {
+          res.redirect('/chat');
+        } else {
+          Message.find({receiver: currentUser.id, sender: usr.id}, (err, messages) => {
```

Comment:
Koliko sam skužio ovdje samo postavljas svim porukama isSeen na true. To možeš lakše da koristiš Message.update(query, update). Posložim ja pa lako pogledas kasnije.